doc/Lorre's Block Fetching Algorithm

```diff
@@ -0,0 +1,96 @@
+Lorre's Block Fetching Algorithm:
+
+The following logic is implemented via the processBlocks function, which is a private helper of the
+more general getBlocks function.
+
+Initially, when Lorre is first run with an empty database, it connects to the head block on the chain
+at level n, and will add blocks at levels 0..n to the chain. If there is no fork at all,
+the next iteration of Lorre will run. If there are no new blocks, nothing is added to the database.
+If there are new blocks, let the head block's level be n_2. Then, blocks n+1, n+2...n_2 will be
+added to the database. Inductively, if there are never any forks, then at iteration i of Lorre,
+Lorre will add blocks n_{i-1} + 1, n_{i-1} + 2...n_i to the database.
+
+ASSUMPTION: In a non-forked chain, there can only be one block at each level. [i]
+
+------------------------------------------------------------------------------------------------------
+
+Lorre's algorithm for fetching blocks that occur as a result of a fork:
+
+The following logic is implemented via the followFork function, which is a private helper of the
+more general getBlocks function.
+
+In the normal case, we place blocks into a blocks table, with the fields being all of the associated
+metadata. In the forked case, we will have an invalidated blocks table to invalidate blocks that are
+not internally consistent with the new chain being recognized by Lorre. Here, the necessary fields are
+block_id (the block's hash), level, and an is_invalidated boolean flag.
```

Comment:
what are the chances of conflicting blocks with the same block hash and level?

Reply:
same block hash will never happen (the probability is astronomically low). If there are two blocks at the same level, a fork must have occurred, hence the need for this function